src/backend/gpopt/gpdbwrappers.cpp

```diff
@@ -2735,4 +2747,43 @@ gpdb::IsTypeRange(Oid typid)
 	return false;
 }
 
+// Check that the index is usable in the current snapshot and if not, save the
+// xmin of the current snapshot. returns true if the index is not usable and
+// should be skipped.
+bool
+gpdb::SetMDCacheTransientState(Relation index_rel)
+{
+	GP_WRAP_START;
+	{
+		bool result =
+			index_rel->rd_index->indcheckxmin &&
+			!TransactionIdPrecedes(
+				HeapTupleHeaderGetXmin(index_rel->rd_indextuple->t_data),
+				TransactionXmin);
+		if (result)
+			mdcache_transaction_xmin = TransactionXmin;
+		return result;
+	}
+	GP_WRAP_END;
+	// ignore index if we can't check it visibility for some reason
+	return true;
+}
+
+void
+gpdb::ResetMDCacheTransientState()
+{
+	mdcache_transaction_xmin = InvalidTransactionId;
+}
+
+bool
+gpdb::IsMDCacheTransient()
```

Comment:
then it is logical to rename
```suggestion
gpdb::IsMDCacheTransientState()
```

Reply:
May be. but how correct is this from the point of view of the English language?

Reply:
May be, according to MDCacheNeedsReset rename new functions to
MDCacheSetTransientState
MDCacheResetTransientState
MDCacheInTransientState
?

Reply:
IsMDCacheTransientStateSet